src/endpoint/s3/s3_utils.js

```diff
@@ -226,9 +226,9 @@ function parse_sse(req) {
         }
     }
 
-    if (algorithm !== 'AES256' && algorithm !== 'aws:kms') throw new S3Error(S3Error.InvalidDigest);
+    if (algorithm !== 'AES256' && algorithm !== 'aws:kms') throw new S3Error(S3Error.InvalidArgument);
```

Comment:
Would you share the link for what you found?

Reply:
aws documentation doesn't specify the error returned only the valid values: https://docs.aws.amazon.com/AmazonS3/latest/API/API_PutObject.html#API_PutObject_RequestSyntax
this is the value returned by aws when I sent invalid request

Reply:
It was in the test itself, thanks.